parser_library/src/context/instruction.cpp

```diff
@@ -21,7 +21,55 @@ using namespace hlasm_plugin::parser_library::context;
 using namespace hlasm_plugin::parser_library::checking;
 using namespace hlasm_plugin::parser_library;
 
-std::string_view instruction::mach_format_to_string(mach_format f)
+namespace {
+// Determines in which systems is the instruction supported
+// First Byte determines instruction support since a specific iteration of the Z system
+// Other bits determine instruction support in other systems
+// Example: 0010 1010 0101:
+//  -   0010 xxxx xxxx -> Instruction supported on ESA systems
+//  -   xxxx 1010 xxxx -> Instruction supported on 370 systems and is also part of universal instruction set
+//  -   xxxx xxxx 0101 -> Instruction supported since Z12 systems up to now
+enum class supported_system
+{
+    NO_Z_SUPPORT = 0,
+    SINCE_ZOP = 1,
+    SINCE_YOP,
+    SINCE_Z9,
+    SINCE_Z10,
+    SINCE_Z11,
+    SINCE_Z12,
+    SINCE_Z13,
+    SINCE_Z14,
+    SINCE_Z15,
+    ESA = 1 << 5,
+    XA = 1 << 6,
+    _370 = 1 << 7,
+    DOS = 1 << 8,
+    UNI = 1 << 9
+};
+
+constexpr supported_system operator|(supported_system a, supported_system b)
```

Comment:
These operations do not always make sense... I think they should. e,g, `SINCE_Z10|SINCE_Z11` should be equal `SINCE_Z11` (like a set union)

Reply:
Alright. I also added some test to verify this behavior

Reply:
However, I think that in case of `SINCE_Z10|SINCE_Z11` it should result in `SINCE_Z10`, right?

Reply:
I guess it depends on the exact semantics?
If I do union of instructions from Z10 and Z11, I get Z11.
If you do union of requirements "need to run on" Z10 and Z11, you get Z10 (which I understood as intersection of instruction sets)

Reply:
As discussed